lib_xua/src/core/endpoint0/xua_ep0_uacreqs.xc

```diff
@@ -513,33 +514,39 @@ int AudioClassRequests_2(XUD_ep ep0_out, XUD_ep ep0_in, USB_SetupPacket_t &sp, c
                                 return result;
                             }
 
-                            /* Check for correct datalength for clock sel */
                             if(datalength == 1)
                             {
-                                if (!isnull(c_clk_ctl))
+                                int clockIndex = (int) (buffer, unsigned char[])[0];
+                                clockIndex -= 1; /* Index to/from from host is 1-based */
+
+                                if((clockIndex > 0) && (clockIndex < CLOCK_COUNT))
```

Comment:
Should this be `clockIndex >= 0` because the 1-based index from the host has been decremented by one?

Reply:
yes! thanks